### PR TITLE
Fix Dockerfile for Alternative Terraform worker image.

### PR DESCRIPTION
**Pull request by Markieta**
## PR Objective

- [x] Fixing inaccurate docs

## Description

This is to adjust the sample Dockerfile to inject the CA certificate properly. The existing sample does not work as expected.

### CODE REVIEW

**Comment by sudomateo** [comment]
> The existing sample does not work as expected.

Would you be able to provide more context here?

Just for reference, I made a comment related to the changes in this pull request [here](https://github.com/hashicorp/terraform-website/pull/1283#discussion_r438484167).

**Comment by Markieta** [comment]
@sudomateo writing the CA trust bundle (custom + OS) directly to `/etc/ssl/certs/ca-certificates.crt` does not work as intended. When the image is being initialized, the following error occurs:

```bash
Error: Failed to request discovery document: Get https://<tfe-host>/.well-known/terraform.json: x509: certificate signed by unknown authority
```

In my case, I have added my own root CA to my system's trust (as per the lock symbol in the address bar) and copied that bundle into the container.

Screenshot from 2020-06-19 20-43-55

While `curl` may work with this method, some commands do not, which may be causing the issue above. For example, with `wget`:

```bash
sudo docker run --rm -it <repo>/<image> /bin/bash

root@faec2b0b1b6e:/# wget https://tfe.markieta.io:8800
--2020-06-21 14:44:43--  https://tfe.markieta.io:8800/
Resolving tfe.markieta.io (tfe.markieta.io)... 192.168.122.233
Connecting to tfe.markieta.io (tfe.markieta.io)|192.168.122.233|:8800... connected.
ERROR: cannot verify tfe.markieta.io's certificate, issued by 'emailAddress=chrismarkieta@gmail.com,CN=markieta.io,OU=DevOps,O=Arctiq,L=Toronto,ST=Ontario,C=CA':
  Unable to locally verify the issuer's authority.
To connect to tfe.markieta.io insecurely, use `--no-check-certificate'.
```

Even though my root CA was included in `/etc/ssl/certs/ca-certificates.crt`.

**Comment by sudomateo** [comment]
Hmm. I was not able to reproduce that. I was able to use this `Dockerfile` to build an alternative worker image.

```
# This Dockerfile builds the image used for the worker containers.
FROM ubuntu:xenial

# Inject the CA certificates.
ADD ca-certificates.crt /etc/ssl/certs/ca-certificates.crt

# Install software used by Terraform Enterprise.
RUN apt-get update && apt-get install -y --no-install-recommends \
    sudo unzip daemontools git-core ssh wget curl psmisc iproute2 openssh-client redis-tools netcat-openbsd
```

The `ca-certificates.crt` file was populated with the root CA certificate for my Terraform Enterprise instance. The entire TLS chain consisted of two TLS certificates; the end-entity certificate presenting by my Terraform Enterprise instance and the root CA certificate that issued that end-entity certificate.

The `curl`, `wget`, and `openssl` commands all work from within the alternative worker image. I am also able to execute Terraform runs.

Either way, I can do some testing this week to verify that the changes proposed in this pull request are sufficient to add to the documentation. Thank you for your work on this!

**Comment by Markieta** [comment]
Thanks for taking a look!

We were able to reproduce the issue in a customer's environment as well as my own lab.

To be clear, we are using locally created/self-signed CA and certs as TFE is in a private network.

**Comment by sudomateo** [comment]
You're welcome!

> To be clear, we are using locally created/self-signed CA and certs as TFE is in a private network.

I was too. I wonder if one of the certificates weren't added. Either way, I'll definitely get to this this week. I appreciate all the information you've provided.

File: content/source/docs/enterprise/install/installer.html.md [hunk 1]
```diff
@@ -143,7 +143,7 @@ TFE runs `terraform plan` and `terraform apply` operations in a disposable Docke
 ### Requirements
  - The base image must be `ubuntu:xenial`.
  - The image must exist on the Terraform Enterprise host. It can be added by running `docker pull` from a local registry or any other similar method.
- - A file containing all necessary PEM-encoded CA certificates must be placed in `/etc/ssl/certs/ca-certificates.crt`. This includes both custom CA certificates and default CA certificates that come with the operating system. The CA certificates configured in the [CA Bundle settings](#certificate-authority-ca-bundle) will not be automatically added to this image at runtime.
+ - A file containing the necessary PEM-encoded CA certificate for the Terraform Enterprise host must be placed in `/usr/local/share/ca-certificates/`. The CA certificates configured in the [CA Bundle settings](#certificate-authority-ca-bundle) will not be automatically added to this image at runtime.
```
Comment (sudomateo) [inline content/source/docs/enterprise/install/installer.html.md:146]:
```suggestion
 - All necessary PEM-encoded CA certificates must placed within the `/usr/local/share/ca-certificates` directory. Each file added to this directory must end with the `.crt` extension. The CA certificates configured in the [CA Bundle settings](#certificate-authority-ca-bundle) will not be automatically added to this image at runtime.
```

File: content/source/docs/enterprise/install/installer.html.md [hunk 2]
```diff
@@ -152,12 +152,15 @@ TFE runs `terraform plan` and `terraform apply` operations in a disposable Docke
...
-# Inject the CA certificates.
-ADD ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
-
 # Install software used by Terraform Enterprise.
 RUN apt-get update && apt-get install -y --no-install-recommends \
-    sudo unzip daemontools git-core ssh wget curl psmisc iproute2 openssh-client redis-tools netcat-openbsd
+    sudo unzip daemontools git-core ssh wget curl psmisc iproute2 openssh-client redis-tools netcat-openbsd ca-certificates
+
+# Include the Terraform Enterprise CA certificate.
+ADD tfeCA.crt /usr/local/share/ca-certificates/tfeCA.crt
```
Comment (sudomateo) [inline content/source/docs/enterprise/install/installer.html.md:160]:
```suggestion
# Include all necessary CA certificates.
ADD example-root-ca.crt /usr/local/share/ca-certificates/
ADD example-intermediate-ca.crt /usr/local/share/ca-certificates/
```

File: content/source/docs/enterprise/install/installer.html.md [hunk 3]
```diff
@@ -152,12 +152,15 @@ TFE runs `terraform plan` and `terraform apply` operations in a disposable Docke
...
+
+# Insert the CA certificate as part of the trust.
```
Comment (sudomateo) [inline content/source/docs/enterprise/install/installer.html.md:162]:
```suggestion
# Update the CA certificates bundle to include newly added CA certificates.
```

**Comment by Markieta** [comment]
Looks great! Just added a missing word^. Thanks for your help @sudomateo

**Pull request merged**